src/pages/markdown/2023-chatgpt-in-daily-work/index.md

```diff
@@ -0,0 +1,85 @@
+---
+path: '/blog/2023-chatgpt-in-daily-work'
+date: '2023-04-05T13:00:00.000Z'
+title: '3 Ways to Use ChatGPT in Your Work as a React Native Software Engineer'
+image: header.jpg
+author: Ilya Kushner
+---
+
+As a React Native software engineer, you're always looking for ways to improve your workflow and be more productive. ChatGPT is a tool that can help you with this goal. It can be used in a variety of ways to make your work easier and more efficient. Here are three ways to use ChatGPT in your work as a React Native software engineer
+
+## 1. Generating Ideas
+
+As a software engineer, you're often tasked with coming up with creative solutions to complex problems. However, it's not always easy to come up with fresh and innovative ideas. ChatGPT can help you generate new ideas by offering suggestions based on your prompts.
+
+To use ChatGPT for idea generation, simply type in a prompt related to your project or problem. For example, if you're stuck on how to improve the performance of your app, you might type in "How can I improve the performance of my React Native app?" ChatGPT will then offer several possible solutions you could try, based on current best practices in the industry.
+
+Here are the few example of prompts:
+
+- When You use for example `react-native-cameraroll` all paths on iOS are in `ph://` format, and to upload such file it's needed to be converted to `file://` path format
+
+![path conversion](./ph-path-prompt.png)
+
+- When You can't use library, for example if it's deprecated, ChatGPT can help You with alternatives:
+
+![library alternatives](./alternatives-example.png)
+
+- Or You can ask it to cover something with unit tests:
+
+![unit test example](./unit-tests-example.png)
+
+- And You can ask it for some ideas when stuck with a problem  
+
+![stuck problem example](./stuck-problem-example.png)
+
+## 2. Code Review
+
+Code review is an essential part of the software development process. However, it can be difficult to catch all the bugs and errors in your own code, especially if you've been working on it for a while. ChatGPT can help you catch mistakes you might have missed by reviewing your code and offering suggestions for improvements.
+
+To use ChatGPT for code review, simply send your code and ask for feedback. ChatGPT will examine your code for errors, potential security vulnerabilities, and areas for optimization. It can also suggest alternative approaches or highlight areas where you could simplify your code. This can help you catch bugs and improve the performance of your app.
+
+Here You can see an example of a code reviewed by ChatGPT:
+
+![code review example](./code-review-example.png)
```

Comment:
Please note that the analysis made by GPT is not fully correct. The code review in the screenshot says the following:

- _The `value` string is then checked using the `RegExp.test()` method to see if it matches the given `regExp` pattern. The `test()` method returns a boolean indicating whether the pattern matches the value or not._
There isn't really any place where program calls the `test()` method explicitly. Also:
<img width="382" alt="Screenshot 2023-04-07 at 11 09 16" src="https://user-images.githubusercontent.com/1140608/230580900-dacb702b-565b-42f3-9e60-bfc182c16156.png">


- _The function attempts to validate if the given 'value' matches the given ' regExp' pattern._
This is not a precise explanation - what the function does (if I understand it correctly) is to check whether the **entire** value matches the pattern.

For example, this code:
```
checkSuccess("abbc", /ab+/)
```
returns `false`. 

With the usual definition of "matching" we would assume that it should return true, somethin like this: `!!"abbc".match(/ab+/) `

The bit with GPT complaining about error checking returning always true is pretty cool though!

Reply:
> The value string is then checked using the RegExp.test() method to see if it matches the given regExp pattern. The test() method returns a boolean indicating whether the pattern matches the value or not.
> There isn't really any place where program calls the test() method explicitly. Also:
I think I should mention it right below the screenshot

>The function attempts to validate if the given 'value' matches the given ' regExp' pattern.
This is not a precise explanation - what the function does (if I understand it correctly) is to check whether the entire value matches the pattern.
In this case given value means entire value, at least I'm for some reason pretty sure in it, and I'm not sure if it should be mentioned below the screenshot...
>

Reply:
And.... noticed now

Reply:
My main takeaway from this example would be something like: GPT can help you analyze some things but I wouldn't rely on it in terms of deep understanding of how something works.

Reply:
It's really good, actually, it shows one more time, that You shouldn't use it as the one and only source of truth and rely on it in everything, that's why I think it's better to keep this example

Reply:
Yes, definitely keep this example! Just add some note about this and point out what's imprecise and it will be great!

Reply:
Yep, added below photo